Network/DNS/IO.hs

```diff
@@ -53,13 +57,21 @@ import Network.DNS.Types
 
 ----------------------------------------------------------------
 
+#if MIN_VERSION_conduit(1,3,0)
+sink :: ConduitT ByteString Void IO (DNSMessage, PState)
```

Comment:
If you want something which will work on both new and old conduit, you can use:

    ConduitM ByteString Void IO (DNSMessage, PState)

Also, you can probably generalize the type signature to:

    ConduitM ByteString o IO (DNSMessage, PState)